File: Java/Tests/jdi-uitest-wintests/src/test/java/com/epam/jdi/uitests/testing/unittests/tests/common/TextFieldTests.java
package com.epam.jdi.uitests.testing.unittests.tests.common;

import com.epam.jdi.uitests.core.interfaces.common.ITextField;
import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.testing.unittests.User;
import com.epam.jdi.uitests.testing.unittests.enums.Preconditions;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Factory;
import org.testng.annotations.Test;

import java.lang.reflect.Method;
import java.util.function.Supplier;

import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.CONTACT_PAGE_FILLED;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDIApplication.contactFormPage;
import static com.epam.jdi.uitests.win.settings.WinSettings.getDriver;
import static org.testng.Assert.assertEquals;

/**
 * Created by Dmitry_Lebedev1 on 11/12/2015.
 */
public class TextFieldTests extends InitTests {
    public String text = User.DEFAULT.name;
    public String contains = "ame";
    public String regex = ".am.";
    Supplier<ITextField> textField = () -> contactFormPage.name;
    private String inputText = "text123!@#$%^&*()";
    private String expectedText = "text123!@#$%^&*()";
    private Preconditions page = CONTACT_PAGE_FILLED;
    private String elementType = "TextField";

    public TextFieldTests() {
    }

    public TextFieldTests(String elementType, Preconditions page, Supplier<ITextField> textItem,
                          String inputText, String expectedText, String text,
                          String contains, String  regex){
        this.elementType = elementType;
        this.page = page;
        this.textField = textItem;
        this.inputText = inputText;
        this.expectedText = expectedText;
        this.text = text;
        this.contains = contains;
        this.regex = regex;
    }

    @BeforeMethod
    public void before(final Method method) {
        isInState(page, method);
    }

    @Test
    public void inputTest() throws Exception {
        textField.get().input(inputText);
        assertEquals(textField.get().getText(), text + expectedText);
    }

    @Test
    public void sendKeyTest() throws Exception {
        textField.get().sendKeys(inputText);
        assertEquals(textField.get().getText(), text + expectedText);
    }

    @Test
    public void newInputTest() throws Exception {
        textField.get().newInput(inputText);
        assertEquals(textField.get().getText(), expectedText);
    }

    @Test
    public void clearTest() throws Exception {
        textField.get().clear();
        assertEquals(textField.get().getText(), "");
    }

    @Test
    public void multiKeyTest() throws Exception {
        for(char letter : inputText.toCharArray())
            textField.get().sendKeys(Character.toString(letter));
        assertEquals(textField.get().getText(), text + expectedText);
    }

    @Test
    public void focusTest() throws Exception {
        String value = "value";
        String attrName = "focusTest";
        textField.get().setAttribute(attrName, value);
        textField.get().focus();
        String resValue = getDriver().switchTo().activeElement().getAttribute(attrName);
        assertEquals(value, resValue);
    }

    @Factory
    public Object[] textTests() {
        return new Object[]{
                new TextTests(elementType, page, textField::get, text, contains, regex)
        };
    }
}

File: Python/Test/jdi_uitests_webtests/test/common/text_field_tests.py
from JDI.core.settings.jdi_settings import JDISettings
from JDI.jdi_assert.testing.assertion import Assert
from Test.jdi_uitests_webtests.main.enums.preconditions import Preconditions
from Test.jdi_uitests_webtests.main.page_objects.epam_jdi_site import EpamJDISite
from Test.jdi_uitests_webtests.test.init_tests import InitTests


class TextFieldTests(InitTests):

    text_field = EpamJDISite.contact_form_page.name_text_field
    input_text = "text123!@#$%^&*()"
    initial_text = "init"

    def setUp(self):
        Preconditions.CONTACT_PAGE.is_in_state()
        self.text_field.input(self.initial_text)

    def test_input(self):
        self.text_field.input(self.input_text)
        Assert.assert_equal(self.text_field.get_text(), self.initial_text+self.input_text)

    def test_send_key(self):
        self.text_field.send_keys(self.input_text)
        Assert.assert_equal(self.text_field.get_text(), self.initial_text+self.input_text)

    def test_new_input(self):
        self.text_field.new_input(self.input_text)
        Assert.assert_equal(self.text_field.get_text(), self.input_text)

    def test_clear(self):
        self.text_field.clear()
        Assert.assert_equal(self.text_field.get_text(), "")

    def test_multi_key(self):
        for letter in self.input_text:
            self.text_field.send_keys(letter)
        Assert.assert_equal(self.text_field.get_text(), self.initial_text+self.input_text)

    def test_focus(self):
        value = "value"
        attribute_name = "focusTest"
        self.text_field.set_attribute(attribute_name, value)
        self.text_field.focus()
        driver = JDISettings.get_driver_factory().get_driver()
        result_value = driver._switch_to.active_element.get_attribute(attribute_name)
        Assert.assert_equal(value, result_value)
